pkgs/by-name/co/compiler-explorer/cfg/utils.nix

```diff
@@ -0,0 +1,47 @@
+{ lib, writeText, runCommand, python3Full }:
+rec {
+  pythonJsonToDot = writeText "jstodot" ''
+    import sys
+    import json
+    def dot_notate(obj, target=None, prefix=""):
+        if target is None:
+            target = {}
+        for key, value in obj.items():
+            if isinstance(value, dict):
+                dot_notate(value, target, prefix + key + ".")
+            else:
+                target[prefix + key] = value
+        return '\n'.join([f"{key}={value}" for key, value in target.items()])
+    try:
+        print(dot_notate(json.loads(sys.stdin.read())))
+    except json.JSONDecodeError as e:
+        print(f'Error parsing JSON: {e}', file=sys.stderr)
+  '';
+
+  attrToDot = x: builtins.readFile (runCommand "jsonToDot.dot" { } ''
```

Comment:
This should be possible to rewrite in Nix (see pkgs.format). I can give this a try.

Reply:
Please do! I'm very much a newcomer, I can give it a try also, this is something I wanted to rewrite this to make nix-only, feel welcomed to work on this :)

Reply:
Invite received! Do you think you can enable issues in your fork, to make it a bit easier to communicate?